### ISSUE TITLE
DIIS extrapolate error with complex TA arrays

### ISSUE THREAD

**Issue by xwang862** [ValeevGroup/tiledarray#107]
My `DIIS` object was templated by `mpqc::lcao::cc::T1T2<std::complex<double>, TA::TensorZ, TA::SparsePolicy>`. When it compiles to the function `DIIS::extrapolate`, thresholds `zero_norm` and `zero_determinant` got the type `std::complex<double>`. And then errors occured:

```
/Users/xiaowang/Programs/tiledarray/install/include/TiledArray/algebra/diis.h:195:33: error: invalid operands to binary expression ('Scalar' (aka 'std::__1::complex<double>') and 'const value_type' (aka 'const std::__1::complex<double>'))
            if (B_(nskip,nskip) > zero_norm)
                ~~~~~~~~~~~~~~~ ^ ~~~~~~~~~
/Users/xiaowang/Programs/mpqc4/mpqc4/src/mpqc/chemistry/qc/cc/gamma_point_ccsd.h:442:14: note: in instantiation of member function 'TiledArray::DIIS<mpqc::lcao::cc::T1T2<std::__1::complex<double>, TiledArray::Tensor<std::__1::complex<double>, Eigen::aligned_allocator<std::__1::complex<double> > >, TiledArray::SparsePolicy> >::extrapolate' requested here
        diis.extrapolate(t, r);
             ^
```

```
/Users/xiaowang/Programs/tiledarray/install/include/TiledArray/algebra/diis.h:226:23: error: invalid operands to binary expression ('value_type' (aka 'std::__1::complex<double>') and 'const value_type' (aka 'const std::__1::complex<double>'))
          if (absdetA < zero_determinant) {
              ~~~~~~~ ^ ~~~~~~~~~~~~~~~~
``` 

Will it be better to use moduli in such comparisons?

**Comment by evaleev** [comment]
@xwang862 please try 8241ea1

**Comment by xwang862** [comment]
No `scalar_type` in `TA::DistArray`...

```
/Users/xiaowang/Programs/tiledarray/install/include/TiledArray/algebra/diis.h:84:27: error: no type named 'scalar_type' in 'TiledArray::DistArray<Tensor<double, Eigen::aligned_allocator<double> >, TiledArray::SparsePolicy>'
      typedef typename D::scalar_type scalar_type;
              ~~~~~~~~~~~~^~~~~~~~~~~
/Users/xiaowang/Programs/mpqc4/mpqc4/src/mpqc/chemistry/qc/scf/rhf_impl.h:147:32: note: in instantiation of template class 'TiledArray::DIIS<TiledArray::DistArray<Tensor<double, Eigen::aligned_allocator<double> >, TiledArray::SparsePolicy> >' requested here
  TiledArray::DIIS<array_type> diis;
                               ^
/Users/xiaowang/Programs/mpqc4/mpqc4/src/mpqc/chemistry/qc/scf/rhf.cpp:28:28: note: in instantiation of member function 'mpqc::lcao::RHF<TiledArray::Tensor<double, Eigen::aligned_allocator<double> >, TiledArray::SparsePolicy>::solve' requested here
template class mpqc::lcao::RHF<TA::TensorD, TA::SparsePolicy>;
                           ^
1 error generated.
```

**Comment by evaleev** [comment]
try 3b2076b5214cac31ce58878f6a815d5b530aa819

**Comment by xwang862** [comment]
it should be fixed by 1fceb55
